src/internals/entity.rs

```diff
@@ -62,20 +58,27 @@ impl<'a> Iterator for Allocate {
 
     #[inline(always)]
     fn next(&mut self) -> Option<Self::Item> {
+        // Get next block whenever the block ends.
         if self.next % BLOCK_SIZE == 0 {
-            // This is either the first block, or we overflowed to the next block.
-            self.next = NEXT_ENTITY.fetch_add(BLOCK_SIZE, Ordering::Relaxed);
-            debug_assert_eq!(self.next % BLOCK_SIZE, 0);
+            static NEXT_ENTITY_BLOCK_START: AtomicU64 = AtomicU64::new(1);
```

Comment:
Initializing this to BLOCK_SIZE has the same effect of preventing the first entity from being allocated as 0, but without causing 1 entity in every block to be skipped. It skips the first block instead (which isn't a problem).